app.rb

```diff
@@ -57,4 +75,79 @@ def list_books
       puts 'You have no books added!'
     end
   end
+
+  def save_files
+    people = []
+    books = []
+    rentals = []
+
+    @people.each do |person|
+      people << if person.is_a? Student
+                  {
+                    age: person.age,
+                    name: person.name,
+                    parent_permission: person.parent_permission
+                  }
+                else
+                  person.is_a? Teacher
+                  {
+                    age: person.age,
+                    name: person.name,
+                    specializatin: person.specializatin
+                  }
+                end
+
+      @books.each do |book|
+        books << {
+          title: book.title,
+          author: book.author
+        }
+      end
+      @rentals.each do |rental|
+        rentals << {
+          date: rental.date,
+          person: rental.person.name,
+          book: rental.book.title
+        }
+      end
+    end
+
+    File.write(@people_path, JSON.generate(people))
+    File.write(@books_path, JSON.generate(books))
+    File.write(@rentals_path, JSON.generate(rentals))
+  end
+
+  def load_people
+    people = JSON.parse(File.read(@people_path))
+    people.each do |person|
+      @people <<
+        if person['specialization']
+          Teacher.new(person['specialization'], person['age'], person['name'])
+        else
```

Comment:
- [ ] Your recent changes seem to have introduced a bug and I can no longer create a teacher, it gives an argument error everytime I add a specialization. Kindly fix this small issue.